__tests__/app.test.js

```diff
@@ -230,6 +230,67 @@ describe("/api/articles", () => {
         });
     });
   });
+
+  describe("PATCH /api/articles/:article_id", () => {
+    test("PATCH 200: Responds with the updated article object", () => {
+      return request(app)
+        .patch("/api/articles/5")
+        .send({ inc_votes: 20 })
+        .expect(200)
+        .then(({ body }) => {
+          const { article } = body;
+
+          expect(article.article_id).toBe(5);
+          expect(article.votes).toBe(20);
+
+          expect(typeof article.title).toBe("string");
+          expect(typeof article.topic).toBe("string");
+          expect(typeof article.author).toBe("string");
+          expect(typeof article.body).toBe("string");
+          expect(typeof article.created_at).toBe("string");
+          expect(typeof article.article_img_url).toBe("string");
+        });
+    });
+
+    test("PATCH 404: Responds with an adequate status and message when provided with a valid but non-existent ID", () => {
+      return request(app)
+        .patch("/api/articles/99999")
+        .send({ inc_votes: 20 })
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("Article not found.");
+        });
+    });
+
+    test("PATCH 400: Responds with an adequate status and message when provided with an invalid id", () => {
+      return request(app)
+        .patch("/api/articles/invalid-id")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+
+    test("PATCH 400: Responds with an adequate status and error message when the data type of the input's property values are not correct", () => {
+      return request(app)
+        .patch("/api/articles/5")
+        .send({ inc_votes: "invalid-data" })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+
+    test("PATCH 400: Responds with an adequate status and error message when provided with an incomplete body", () => {
+      return request(app)
+        .patch("/api/articles/5")
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+  });
```

Comment:
^^ I think this last test should just respond as per usual - i.e. it should respond with the comment! Ultimately, nothing wrong has happened, so it's not an error :)

Reply:
I thought that it would trigger an error since the patching did not happen? Because there are missing values, the query throws a 23502 error (error: null value in column "votes" of relation "articles" violates not-null constraint). 

I guess I could check if the body has no keys before querying in the model; if that is the case, I can just query the db for the article with that id to return the unpatched article. And then, if the body _has_ keys, just query as intended to update the votes? Would that be better?